en/02_Developer_Guides/18_Cookies_And_Sessions/02_Sessions.md

```diff
@@ -100,9 +100,52 @@ including form and page comment information. None of this is vital but `clear_al
 $session->clearAll();
 ```
 
-## Cookies
+## Configuration
 
-### Samesite attribute
+### Session lifetime
+
+By default, the session cookie will only last until the user closes their browser, and the session on the server-side will live for the number of seconds set in the [`session.gc_maxlifetime`](https://www.php.net/manual/en/session.configuration.php#ini.session.gc-maxlifetime) ini configuration since it was last touched.
+
+You can configure the session lifetime by setting the [`Session.timeout`](SilverStripe\Control\Session->timeout) configuration property to the number of seconds the session should be valid for since the last modified time.
+
+```yml
+SilverStripe\Control\Session:
+  timeout: 86400
+```
+
+> [!WARNING]
+> If you use a save handler which isn't included in `silverstripe/framework`, it may treat the session lifetime differently. You should check its docuemntation or implementation to understand how it works.
+
+### Save handler
+
+You can choose how session are handled by setting the [`Session.save_handler`](api:SilverStripe\Control\Session->save_handler) configuration property. By default this is set to [`FileSessionHandler`](api:SilverStripe\Control\SessionHandler\FileSessionHandler).
+
+If you want to use the default PHP file session handler instead, you can set the `Session.save_handler` configuration to `null`.
+
+```yml
+SilverStripe\Control\Session:
+  save_handler: null
+```
+
+You can also set it to the FQCN or injector service name of any session handler that implements [`SessionHandlerInterface`](https://www.php.net/manual/en/class.sessionhandlerinterface.php).
+
+#### `FileSessionHandler`
+
+The default file-based session handler for PHP holds a lock on the session file while the session is open. In practice this means that multiple concurrent requests from the same user have to wait for one another to finish processing after a session has been started. This includes AJAX requests.
+
+To resolve this problem, Silverstripe CMS comes with [`FileSessionHandler`](api:SilverStripe\Control\SessionHandler\FileSessionHandler).
+
+Note that in edge case scenarios, for example if your application wants to modify a session value *based on the value that is already set* and must do so for each request, non-blocking sessions may cause unexpected results.
+
+`FileSessionHandler` differs from the default PHP file session handler in the following ways:
+
+1. It doesn't lock the session file, and therefore doesn't block concurrent requests.
```

Comment:
```suggestion
1. It doesn't keep a lock on the session file, and therefore doesn't block concurrent requests.
```
I think this is correct? It does lock the file, it's just it unlocks it again when it's finished, which is practive happens very quickly

Reply:
It doesn't lock the file, or if it does that's an implementation detail of PHP or of the filesystem and not of the session handler.